### ISSUE TITLE
Adding `+` to version prevents checkout

### ISSUE THREAD

**Issue by jodator** [pnpm/action-setup#140]
When the `packageManager` version is defined in a corepack compatible way, e.g. `pnpm:9.5.0+` the action checkout fails with

```
ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  pnpm@9.5.0+ isn't supported by any available resolver.
```

It would be good to support `+` range specifier to:
1. Allow devs use any compatible PNPM version (e.g. latest)
2. Make action checkout work with define one `9.5.0`

**Comment by MC-XiaoHei** [comment]
事实上，您可以直接在action中添加
```
        with:
          version: latest
```

**Comment by jodator** [comment]
> actually, you can just add this to workflow

Good to know, but I don't think it resolves the issue of having `9.5.0+` in the `packageManager` field. 

AFAICS this is allowed by `package.json` [spec](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/package.json#L736) but I couldn't find definite answer in [core pack docs](https://nodejs.org/api/packages.html#packagemanager).

**Comment by MC-XiaoHei** [comment]
Yes, I agree that 9.5.0+ should be supported according to the spec (in idea it is also legitimate
